std-bits/table/src/main/java/org/enso/table/parsing/TypeInferringParser.java

```diff
@@ -24,12 +26,26 @@ public TypeInferringParser(
     this.fallbackParser = fallbackParser;
   }
 
+  @Override
+  public Object parseSingleValue(String text, ProblemAggregator problemAggregator) {
+    SimplifiedProblemAggregator internal = new SimplifiedProblemAggregator();
+    for (IncrementalDatatypeParser parser : baseParsers) {
+      internal.reset();
```

Comment:
Is there any particular reason for making `SimplifiedProblemAggregator` mutable? You could easily create a new instance  for each loop iteration. That can't be that bad, no?

Reply:
I think you are right. I wanted to avoid too many allocations, as suggested in another part of review. But I don't think it's a big problem, the JIT should deal with it fine.